code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaSession.java

```diff
@@ -278,9 +280,18 @@ private void trySendHit() {
                             synchronized (mutex) {
                                 boolean shouldRetry = false;
 
-                                if (hitIsSessionStart
+                                if (deviceOffline) {
+                                    // `trySendHit` verifies network connectivity before sending a
+                                    // hit. If the network goes
+                                    // offline after verification but before sending the hit, retry
+                                    // the attempt regardless
+                                    // of the hit type. `trySendHit` will automatically resume when
+                                    // the network connection is
+                                    // restored.
+                                    shouldRetry = true;
+                                } else if (hitIsSessionStart
```

Comment:
Should we add a `debug` log on line 305 when `shouldRetry` is true to state we will retry the hit later?
```
if (!shouldRetry) {
    removeHit();
} else {
   Log.debug("Will attempt to retry sending %s hit later.", hitEventType);
}